### PR TITLE
Updating WNPRC test study metadata

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
This story address two issues 
1. Generally in EHR tests, query validation is done after the study import. This ensures all queries saved in code are still valid and no new changes have broken them.  We found the wnprc tests have actually been doing the study import and query validation before enabling the wnprc_ehr module so the wnprc specific queries have not been getting validated (just core EHR queries being validated). This has been happening for some time, so there are many queries that do not validate. The validation is an important part of automated testing to give confidence any changes have not affected the underlying queries of much of the functionality. 

2. Updating the study dataset XML and test server setup to more closely match what is on WNPRC server. This is mostly setting up the dataset metadata, project structure and linked schemas. This should give greater confidence in current tests and better testability for future tests.

NOTE: The full query cleanup is not done, so query validation is not active yet for these tests. There are still about 10 queries that need clarification on how to make them pass validation.

#### Related Pull Requests
https://github.com/LabKey/testAutomation/pull/1852

#### Changes
* Updated datasets_metadata
* Add new projects/folders to match those needed for queries
* Add linked schemas where necessary
* Some query and query metadata cleanup

### CODE REVIEW

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java [hunk 1]
```diff
@@ -193,36 +204,56 @@ protected File getStudyPolicyXML()
         return TestFileUtils.getSampleData("wnprc_ehr/wnprcEhrTestStudyPolicy.xml");
     }
 
+    private static void folderSetup()
+    {
+
+    }
+
```
Comment (labkey-danield) [inline WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java:211]: Don't keep empty methods.

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java [hunk 2]
```diff
@@ -265,12 +314,52 @@ private void uploadBillingDataAndVerify() throws Exception
...
+    {
+        InsertRowsCommand insertCmd = new InsertRowsCommand("ehr", "project");
+
+        Map<String,Object> rowMap = new HashMap<>();
+        rowMap.put("project", PROTOCOL_PROJECT_ID);
+        rowMap.put("name", PROTOCOL_PROJECT_ID);
+        rowMap.put("protocol", PROTOCOL_ID);
+        rowMap.put("account", ACCOUNT_ID_1);
+        rowMap.put("investigatorId", "1");
+        insertCmd.addRow(rowMap);
```
Comment (labkey-danield) [inline WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java:331]: <nit-pick>
Alternatively you could do:

insertCmd.addRow(Map.of(
"project", PROTOCOL_PROJECT_ID,
"name", PROTOCOL_PROJECT_ID,
"protocol", PROTOCOL_ID,
"account", ACCOUNT_ID_1,
"investigatorId", "1"));

Nothing wrong with the way it is, just an alternative way to do it.

File: WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java [hunk 3]
```diff
@@ -351,15 +510,26 @@ private void createCoreLinkedSchema()
...
+//    {
+//        ListDefinition listDef = new IntListDefinition("Labfee_NoChargeProjects", "key");
+//        listDef.addField(new FieldDefinition("project", FieldDefinition.ColumnType.Integer));
+//        listDef.addField(new FieldDefinition("startDate", FieldDefinition.ColumnType.DateAndTime));
+//        listDef.addField(new FieldDefinition("dateDisabled", FieldDefinition.ColumnType.DateAndTime));
+//        listDef.addField(new FieldDefinition("Createdb", FieldDefinition.ColumnType.Integer));
+//        listDef.addField(new FieldDefinition("Notes", FieldDefinition.ColumnType.String));
+//        listDef.getCreateCommand().execute(createDefaultConnection(), getProjectName());
+//    }
+
```
Comment (labkey-danield) [inline WNPRC_EHR/test/src/org/labkey/test/tests/wnprc_ehr/WNPRC_EHRTest.java:568]: Any reason you are keeping this around? If not remove it, if so add a comment as to why.

**Review by labkey-danield — APPROVED** [review]
The test file looks ok, nothing major. 
I cannot really comment on the data. 

One more comment...
There are two failures in the WNPRC_EHR run on TeamCity for this branch. One of them is a memory leak. You should have Marty, or one of the other devs, review that before merging.

**Comment by labkey-sweta** [comment]
> The test file looks ok, nothing major. I cannot really comment on the data.
> 
> One more comment... There are two failures in the WNPRC_EHR run on TeamCity for this branch. One of them is a memory leak. You should have Marty, or one of the other devs, review that before merging.

Your comments are mostly for the code written by @labkey-martyp, i will let him address them.

**Review by labkey-martyp — APPROVED** [review]

**Pull request merged**